robottelo/cli/hammer.py

```diff
@@ -38,11 +40,37 @@ def _csv_reader(output):
             yield row
 
 
+def _normalize_header(header):
+    """Replace empty spaces with '-' and lower all chars"""
+    return header.replace(' ', '-').lower()
+
+
+def parse_json(stdout):
+    """Parse JSON output from Hammer CLI and convert it to python dictionary
+    while normalizing keys."""
+    parsed = json.loads(stdout)
+    return _normalize_obj(parsed)
+
+
+def _normalize_obj(obj):
+    """Normalize all dct's keys replacing empty spaces with "-" and lowering
+    chars"""
+    if isinstance(obj, dict):
+        return {_normalize_header(k): _normalize_obj(v) for k, v in
+                obj.items()}
+    elif isinstance(obj, list):
+        return [_normalize_obj(v) for v in obj]
+    # doing this to conform to csv parser
+    elif isinstance(obj, int) and not isinstance(obj, bool):
+        return unicode(obj)
+    return obj
+
+
 def parse_csv(output):
     """Parse CSV output from Hammer CLI and convert it to python dictionary."""
     reader = _csv_reader(output)
     # Generate the key names, spaces will be converted to dashes "-"
-    keys = [header.replace(' ', '-').lower() for header in next(reader)]
+    keys = tuple(_normalize_header(header) for header in next(reader))
```

Comment:
Fair enough. In this particular case using a mutable or immutable object don't really matter since the object is used on the function body only

We can evolve the conversation and say that list comprehensions are faster than converting to tuple:

```
In [1]: %timeit [i for i in range(10)]
The slowest run took 14.06 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 1.19 µs per loop

In [2]: %timeit tuple(i for i in range(10))
The slowest run took 6.38 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 1.86 µs per loop
```

But as you can see in this particular case that will not have any big impact since this method will called just in a few times and both cases run really fast.

I've asked just to make sure I was not missing something.